File: barney.js
isSad = true;

function sad() {
    if(isSad) {
        isSad = false;
        console.log("When I get sad,");
        return true;
    } else {
		return {
            "stop" : function () {
                console.log("I stop being sad");
            }
        }
    }
}

function beAwesome() {
    console.log("and be awesome instead");   
}

if (sad() === true) {
    sad().stop();
    beAwesome();
}

File: barney.rb
$isSad = true

class SadBarney
    def stop()
        puts "I stop being sad"
    end
end

def sad()
    if $isSad
        $isSad = false
        puts "When I get sad,"
        return true
    else
        return SadBarney.new
    end
end

def beAwesome()
    puts "and be awesome instead"
end

if sad
    sad.stop
    beAwesome
end
